lib/elixir/lib/io/ansi/docs.ex

```diff
@@ -366,6 +366,10 @@ defmodule IO.ANSI.Docs do
     process_fenced_code_block(rest, text, indent, options, _delimiter = "~~~")
   end
 
+  defp process(["<!--" <> _line | rest], text, indent, options) do
+    process(rest, text, indent, options)
```

Comment:
@wojtekmach I think we should assume it can be multiline or even single line (`<!-- foo --> bar`), so we should handle that too. I think this will do it:

```suggestion
  defp process(["<!--" <> line | rest], text, indent, options) do
    process(drop_comment([line | rest]), text, indent, options)
```

where

```
defp drop_comment([line | rest]) do
  case :binary.split(line, "-->") do
    [_] -> drop_comment(rest)
    [_, line] -> [line | rest]
  end
end

defp drop_comment([]) do
  []
end
```

or something like that.